engine/execution/ingestion/engine.go

```diff
@@ -1155,9 +1159,14 @@ func (e *Engine) saveDelta(ctx context.Context, executionStateDelta *messages.Ex
 		log.Fatal().Err(err).Msg("fatal error while processing sync message")
 	}
 
-	if !bytes.Equal(executionReceipt.ExecutionResult.FinalStateCommit, executionStateDelta.EndState) {
+	finalState, ok := executionReceipt.ExecutionResult.FinalStateCommitment()
+	if !ok {
+		log.Error().Msg("could not get final state: no chunks found")
```

Comment:
```suggestion
		finalState = executionStateDelta.StartState
```

Let's use the delta's start state here, instead of printing a log. Otherwise, the next check will always fail.

If the result doesn't not have any chunk, and StartState and EndState doesn't match, the next line will fail anyway